syntax = "proto2";

package lm;

message connected_msg
{
	required int32     cmd = 1;  // ID
	required int32     from_id = 2;  // ID
	required int32     recv_id = 3;  // str
	required string    msg = 4;  // str
}

message net_msg
{
	required int32    v = 1;  		// version
	required int32    m_id = 2;		// m_id
	required int32    s_id = 3;  	// s_id
	required int32    opt = 4;  	
	optional string   data = 5;		//optional field
}

message chat_msg
{
	required int32     from_id = 1;  // ID
	required int32     recv_id = 2;  // str
	required string    msg = 3;  // str
}

message join_room_msg
{
	required int32     u_id = 1;  // u_id
	required string    msg = 2;  // str
}

message leave_room_msg
{
	required int32     u_id = 1;  // ID
	required string    msg = 2;  // str
}

message sit_msg
{
	required int32     u_id = 1;  // u_id
	required string    msg = 2;  // str
}

message UserInfo_msg
{
	required int32     u_age = 1;			// u_id
	required int32     u_sex = 2;			// u_id
	required string    u_name = 3;			// u_id
	required string    u_address = 4;		// str
}

